paramnb/util.py

```diff
@@ -0,0 +1,23 @@
+import sys
+from collections import OrderedDict
+
+if sys.version_info.major == 3:
+    unicode = str
+    basestring = str
+
+def named_objs(objlist):
+    """
+    Given a list of objects, returns a dictionary mapping from
+    string name for the object to the object itself.
+    """
+    objs = OrderedDict()
+    for k, obj in objlist:
+        if hasattr(k, '__name__'):
+            k = k.__name__
+        elif sys.version_info < (3,0) and isinstance(k, basestring) and not isinstance(k, unicode):
+            k = unicode(k.decode('utf-8'))
+        else:
+            k = unicode(k)
+        objs[k] = obj
```

Comment:
I'm no unicode guru, and (even if I wrote this :-) I have no idea what the elif and else code does.  Can you please make that into a function and give it a useful name so that this function will become readable?  I would assume whatever decoding it's doing isn't specific to named_objs, just working with unicode and non-unicode strings.